Mathlib/FieldTheory/Differential/Basic.lean

```diff
@@ -89,4 +87,87 @@ lemma _root_.algebraMap.coe_logDeriv {F K : Type*} [Field F] [Field K] [Differen
     [Differential K] [Algebra F K] [DifferentialAlgebra F K]
     (a : F) : logDeriv a = logDeriv (a : K) := (logDeriv_algebraMap a).symm
 
+/--
+The differential of a separable value in a field extension.
+-/
+noncomputable def implicitDeriv' (F : Type*) {K : Type*} [CommRing F] [Differential F] [Field K]
+    [Algebra F K] (x : K) : K :=
+  - (aeval x (mapCoeffs (minpoly F x))) / (aeval x (derivative (minpoly F x)))
+
+variable {F : Type*} [Field F] [Differential F] [CharZero F]
+
+noncomputable instance (p : F[X]) [Fact (Irreducible p)] [Fact p.Monic] :
+    Differential (AdjoinRoot p) where
+  deriv := Derivation.liftOfSurjective (f := (AdjoinRoot.mk p).toIntAlgHom) AdjoinRoot.mk_surjective
+    (d := implicitDeriv <| AdjoinRoot.modByMonicHom Fact.out <|
+      implicitDeriv' F (AdjoinRoot.root p)) (fun x hx ↦ by
+      simp_all only [RingHom.toIntAlgHom_apply, AdjoinRoot.mk_eq_zero]
+      obtain ⟨q, rfl⟩ := hx
+      simp only [Derivation.leibniz, smul_eq_mul]
+      apply dvd_add (dvd_mul_right ..)
+      apply dvd_mul_of_dvd_right
+      rw [← AdjoinRoot.mk_eq_zero]
+      unfold implicitDeriv implicitDeriv'
+      simp only [ne_eq, show p ≠ 0 from Irreducible.ne_zero Fact.out, not_false_eq_true,
+        AdjoinRoot.minpoly_root, show p.Monic from Fact.out, Monic.leadingCoeff, inv_one, map_one,
+        mul_one, AdjoinRoot.aeval_eq, Derivation.coe_add, Derivation.coe_smul, Pi.add_apply,
+        Pi.smul_apply, Derivation.restrictScalars_apply, derivative'_apply, smul_eq_mul, map_add,
+        map_mul, AdjoinRoot.mk_leftInverse Fact.out _]
+      rw [div_mul_cancel₀, add_neg_cancel]
+      simp only [ne_eq, AdjoinRoot.mk_eq_zero]
+      have : 0 < p.natDegree := Irreducible.natDegree_pos (Fact.out)
+      apply not_dvd_of_natDegree_lt
+      · intro nh
+        simp [natDegree_eq_zero_of_derivative_eq_zero nh] at this
+      apply natDegree_derivative_lt
+      exact Nat.not_eq_zero_of_lt this
+    )
+
+instance (p : F[X]) [Fact (Irreducible p)] [Fact p.Monic] :
+    DifferentialAlgebra F (AdjoinRoot p) where
+  deriv_algebraMap a := by
+    change (Derivation.liftOfSurjective _ _) ((AdjoinRoot.mk p).toIntAlgHom (C a)) = _
+    rw [Derivation.liftOfSurjective_apply, implicitDeriv_C]
+    rfl
+
```

Comment:
Don't you want `rfl` lemmas that give a formula for the value of the derivation, especially at the adjoint root?